src/Microsoft.OData.Edm/EdmStructuralPropertyAlias.cs

```diff
@@ -0,0 +1,62 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EdmStructuralPropertyAlias.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents an EDM structural (i.e. non-navigation) property with an alias.
+    /// </summary>
+    public class EdmStructuralPropertyAlias : EdmProperty, IEdmStructuralPropertyAlias
+    {
+        private string alias;
+        private IEnumerable<string> path;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmStructuralPropertyAlias"/> class.
+        /// </summary>
+        /// <param name="declaringType">The type that owns this property.</param>
+        /// <param name="name">Name of the property.</param>
+        /// <param name="type">The type of the property.</param>
+        /// <param name="alias">The property alias.</param>
+        /// <param name="path">The path to the referenced property.</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "string", Justification = "defaultValue might be confused for an IEdmValue.")]
+        public EdmStructuralPropertyAlias(IEdmStructuredType declaringType, string name, IEdmTypeReference type, string alias, IEnumerable<string> path)

```

Comment:
With all of this being `public`, we should probably validate everything. For example, don't we need to check the consistency of `declaringType`, `name`, and `type`? And shouldn't `name` always be the last element of `path`? And shouldn't `path` always have at least two elements?

Reply:
Most of the validation are done when the `base` gets called. Like the `declaringType` and `type`. The `name` is the first element of the path. This is the `complex property` in the entity type that acts like a `base` for the path to the `type` that has the `primitive type` which will be used as key for the entity type.